client/app/components/Pages/Profile/Profile.js

```diff
@@ -106,19 +108,41 @@ class Profile extends React.Component {
     }
 
     updateValue(field, newVal) {
-        // TODO: Some Form validation based on field
+        // TODO: Verify email and phone existance.
         var basicInfoCopy = Object.assign({}, this.state.basicInfo);
+        if(basicInfoCopy[field]==newVal){
+            ToastStore.warning("Current "+field+". Please try another one");
+            return; //^ If old value equals updated value, displays appropriate error
+        }
+        if(field=="phone"){
+            if(String(newVal).length!=10 || isNaN(Number(newVal))){
+                ToastStore.warning("Invalid Phone Number. Please try another one");
+                return; //^ If value isn't a numeric, or is not 10 digits long
+            }
+        }
+        else if(field=="email"){
+            if(!validator.validate(newVal)){
+                ToastStore.warning("Invalid Email ID. Please try another one");
+                return; //^ If email ID isn't of format - {x@y.z}
+            }
+        }
+        else if(field=="dob"){
+            var givenDob=new Date(newVal);
+            var presentDate=new Date();
+            if(presentDate-givenDob<16*365*24*60*1000 || presentDate-givenDob>65*365*24*60*1000){
+                ToastStore.warning("Invalid Date of Birth. Please try another one");
+                return; //^ If user is less than 16 years or greater than 65 years
+            }
+        }
```

Comment:
For each of the ToastStore Warnings, the page needs to be reloaded after the warning. This can be done by setting an initial state on render and returning to that initial state after the warnings.

Reply:
But the state isn't modified, is it? because the validation is done before the request's being sent.

Reply:
After any invalid entry, the form retains the previous values. So, why the reload?

Reply:
Let's say I enter a wrong number and click Save. The ToastStore warning is shown but you will see that the page will remain with this new number. The request isn't sent and the value isn't updated, but the field-value will remain on that page after the warning. But after the warning, we need the value to return to the original value of the field, and that state must reload. 
<img width="964" alt="screenshot 2019-02-02 at 7 12 04 pm" src="https://user-images.githubusercontent.com/32671846/52174449-a0b28d80-27b9-11e9-9737-ba8fccde016a.png">

Reply:
Let's say I enter a wrong number and click Save. The ToastStore warning is shown but you will see that the page will remain with this new number. The request isn't sent and the value isn't updated, but the field-value will remain on that page after the warning. But after the warning, we need the value to return to the original value of the field, and that state must reload.
<img width="1792" alt="screenshot 2019-02-03 at 1 41 44 pm" src="https://user-images.githubusercontent.com/32671846/52174453-b6c04e00-27b9-11e9-859e-89b698022981.png">